platforms/ide/tooling-api/src/testFixtures/groovy/org/gradle/integtests/tooling/fixture/ProgressEvents.groovy

```diff
@@ -160,7 +160,21 @@ class ProgressEvents implements ProgressListener {
                 assert event.result.startTime == startEvent.eventTime
                 assert event.result.endTime == event.eventTime
             } else if (event instanceof TestOutputEvent || event instanceof TestMetadataEvent) {
-                // Do nothing - these are treated differently, their contents are stored separately from the events
+                def descriptor = event.descriptor
+                assert seen.add(descriptor)
+                assert !running.containsKey(descriptor)
+
+                // These are progress events and should have a parent
+                assert descriptor.parent != null
+                // parent should also be running
+                assert running.containsKey(descriptor.parent)
+                def parent = operations.find { it.descriptor == descriptor.parent }
+
+                Operation operation = newOperation(event, parent, descriptor)
+                operations.add(operation)
+
+                assert descriptor.displayName == descriptor.toString()
+                assert event.displayName.matches("\\Q${descriptor.displayName}\\E[ \\w-]+")
```

Comment:
❗ I don't think we want to put restrictions on the display name like this.  What if there are Regex control elements in the string?  Wouldn't this allow for regex-injection into an assertion as an attack vector?